src/metrics.cpp

```diff
@@ -267,10 +287,15 @@ int printStats(bool mining)
     auto localsolps = GetLocalSolPS();
 
     if (IsInitialBlockDownload(Params())) {
-        int netheight = currentHeadersHeight == -1 || currentHeadersTime == 0 ? 
-            0 : EstimateNetHeight(params, currentHeadersHeight, currentHeadersTime);
-        int downloadPercent = height * 100 / netheight;
-        std::cout << "     " << _("Downloading blocks") << " | " << height << " / ~" << netheight << " (" << downloadPercent << "%)" << std::endl;
+       if (fReindex) {
```

Comment:
This is undefined behaviour because it accesses `fReindex` without being locked on `cs_main` (which is how other accesses to `fReindex` are locked). Move it to the block above that is locked on `cs_main, cs_vNodes`. (Blocking comment.)

Reply:
I see,  I'll fix it.`